### ISSUE TITLE
Getting two decodings for the same dataitem

### ISSUE THREAD

**Issue by ifsnop** [CroatiaControlLtd/asterix#134]
When decoding cat01 tod, you need to use the MSB byte from the latest cat02 north crossing tod.

In order to get the bytes, one possible solution could be to use two dataitemformat definitions, one with the tod decoded in seconds and the other in hexadecimal.

Will you agree with an implementation that supports this new type of definitions?

Do you think that it could be implemented easily?

~~~
<DataItem id="030">
        <DataItemName>Time of Day</DataItemName>
        <DataItemDefinition>Absolute time stamping expressed as UTC time.</DataItemDefinition>
        <DataItemFormat desc="Three-octet fixed length Data Item in hexadecimal.">
            <Fixed length="3">
                <Bits from="24" to="1" encode="hex">
                    <BitsShortName>ToDHeX</BitsShortName>
                    <BitsName>Time of day in hexadecimal</BitsName>
                </Bits>
            </Fixed>
        </DataItemFormat>
        <DataItemFormat desc="Three-octet fixed length Data Item.">
            <Fixed length="3">
                <Bits from="24" to="1">
                    <BitsShortName>ToD</BitsShortName>
                    <BitsName>Time of day</BitsName>
                    <BitsUnit scale="0.0078125">s</BitsUnit>
                </Bits>
            </Fixed>
        </DataItemFormat>
</DataItem>
~~~

**Comment by dsalantic** [comment]
I see your point, but I'm afraid that proposed solution will make too much confusion, because XML should actually be used to define input format, and not the output format. 

My proposal is to display both vales (hex value and value in specified (and scaled) format) for all variables.
E.g. :    
 'ToD': {'desc': 'Time of day', 'val': 0.0078125, 'hex': 'ABCDEF'}

That way, we will get little longer output for all items, but maybe we could add a command line switch to enable it or not.

I believe that this solution could be easily implemented.
Let me know your opinion.

**Comment by ifsnop** [comment]
That was my idea when proposing #115, but I didn't find an easy way to implement it. 

I don't know how to do it when there are several bits in a dataitem (for example I048/20, TYP/SIM/...), but maybe I'm going to far. 

For me having hex representation of the whole dataitem when it is made of bytes its enough.

**Comment by dsalantic** [comment]
Will be fixed with #135 Implement extensive JSON output format

**Comment by dsalantic** [comment]
Please take a lok at branch "extensive_json_format"  where I implemented quick solution.
feel free to change it and later we can merge it to master if it's OK.
